code/modules/unit_tests/thief.dm

```diff
@@ -0,0 +1,11 @@
+/// Test thiefs for valid json
+/datum/unit_test/thief/Run()
+	load_strings_file(THIEF_FLAVOR_FILE)
+	var/list/all_thief_flavors = GLOB.string_cache[THIEF_FLAVOR_FILE]
+	for(var/list/thief_flavor as anything in all_thief_flavors)
+		try
+			var/objective_path = text2path(thief_flavor["objective_type"])
+			var/datum/objective/exists allocate(objective_path)
```

Comment:
You don't need to do this anyway (or have the try/catch). `text2path` returns null if it doesn't exist.